django/cantusdb_project/main_app/views/chant.py

```diff
@@ -869,6 +873,40 @@ def get_success_url(self):
         return reverse("source-edit-chants", args=[self.object.source.id])
 
 
+class CISearchView(TemplateView):
+    """Search in CI and write results in get_context_data
+    Shown on the chant create page as the "Input Tool"
+    """
+
+    template_name = "ci_search.html"
+
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+        context["genres"] = list(
+            Genre.objects.all().order_by("name").values("id", "name")
+        )
+        search_term: str = kwargs["search_term"]
+        search_term: str = search_term.replace(" ", "+")  # for multiple keywords
+
+        text_search_results: Optional[list[dict]] = get_ci_text_search(search_term)
```

Comment:
This type doesn't agree with the `get_ci_text_search` output type? The type output of the function suggests you'd need to check for `None`s in the `for` loop below.

Reply:
Ah, yes!